src/Templates/AbstractTemplateCompatibility.php

```diff
@@ -0,0 +1,157 @@
+	 *
+	 * @param array         $parsed_block The block being rendered.
+	 * @param array         $source_block An un-modified copy of $parsed_block, as it appeared in the source content.
+	 * @param WP_Block|null $parent_block If this is a nested block, a reference to the parent block.
+	 *
+	 * @return array
+	 */
+	abstract public function update_render_block_data( $parsed_block, $source_block, $parent_block );
+
+	/**
+	 * Inject hooks to rendered content of corresponding blocks.
+	 *
+	 * @param mixed $block_content The rendered block content.
+	 * @param mixed $block         The parsed block data.
+	 * @return string
+	 */
+	abstract public function inject_hooks( $block_content, $block );
+
+	/**
+	 * The hook data to inject to the rendered content of blocks. This also
+	 * contains hooked functions that will be removed by remove_default_hooks.
+	 *
+	 * The array format:
+	 * [
+	 *   <hook-name> => [
+	 *     block_name => <block-name>,
+	 *     position => before|after,
+	 *     hooked => [
+	 *       <function-name> => <priority>,
+	 *        ...
+	 *     ],
+	 *  ],
+	 * ]
+	 * Where:
+	 * - hook-name is the name of the hook that will be replaced.
+	 * - block-name is the name of the block that will replace the hook.
+	 * - position is the position of the block relative to the hook.
+	 * - hooked is an array of functions hooked to the hook that will be
+	 *   replaced. The key is the function name and the value is the
+	 *   priority.
+	 */
+	abstract protected function set_hook_data();
+
+
+	/**
+	 * Remove the default callback added by WooCommerce. We replaced these
+	 * callbacks by blocks so we have to remove them to prevent duplicated
+	 * content.
+	 */
+	protected function remove_default_hooks() {
+		foreach ( $this->hook_data as $hook => $data ) {
+			if ( ! isset( $data['hooked'] ) ) {
+				continue;
+			}
+			foreach ( $data['hooked'] as $callback => $priority ) {
+				remove_action( $hook, $callback, $priority );
+			}
+		}
+
+		/**
+		 * When extensions implement their equivalent blocks of the template
+		 * hook functions, they can use this filter to register their old hooked
+		 * data here, so in the blockified template, the old hooked functions
+		 * can be removed in favor of the new blocks while keeping the old
+		 * hooked functions working in classic templates.
+		 *
+		 * Accepts an array of hooked data. The array should be in the following
+		 * format:
+		 * [
+		 *   [
+		 *     hook => <hook-name>,
+		 *     function => <function-name>,
+		 *     priority => <priority>,
+		 *  ],
+		 *  ...
+		 * ]
+		 * Where:
+		 * - hook-name is the name of the hook that have the functions hooked to.
+		 * - function-name is the hooked function name.
+		 * - priority is the priority of the hooked function.
+		 *
+		 * @since 9.5.0
+		 * @param array $data Additional hooked data. Default to empty
+		 */
+		$additional_hook_data = apply_filters( 'woocommerce_blocks_hook_compatibility_additional_data', array() );
```

Comment:
I'm not sure if I'm understanding correctly what this filter does, but I find the name a bit confusing. Would something like `woocommerce_blocks_hooks_to_skip_in_blocks` better describe what it does? (Probably there are better names, but just trying to see if I understand what it does)

In case we aren't sure about the name, I would mark it as `__experimental` so we can iterate on it in the future.

Reply:
This filter has been introduced by [this PR](https://github.com/woocommerce/woocommerce-blocks/pull/8172/files#diff-02e3c036540a6c7806ce8f3ca2cb1bbcd3d1d1ce67400012c23dd066f87786a9R296).

Reply:
Oh, my bad.  :sweat_smile:  Let's move ahead with this name, then.